wagtail_bucketav/testapp/migrations/0001_initial.py

```diff
@@ -0,0 +1,265 @@
+# Generated by Django 4.0.7 on 2022-08-24 13:32
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+import taggit.managers
+from wagtail import VERSION as WAGTAIL_VERSION
+import wagtail.images.models
+
+try:
+    from wagtail.models import collections
+except ImportError:
+    # Wagtail < 3
+    from wagtail.core.models import collections
+
+import wagtail.search.index
+
+
+def get_wagtailcore_dependency():
```

Comment:
Question: With these, is it possible to just choose the earliest common denominator rather than switching like this?

Reply:
I don't follow - don't we need to provide the minimum dependency for each app, for each version of Wagtail (that has a different set of dependencies)? E.g. we can't specify wagtailcore 0066 as the dependency for a post wagtail 3 test setup, as wagtail 3 requires the `log_entry_jsonfield` migration to have been run, and vice versa.

Reply:
With the work I've been doing on packages it say we just need to use the last migration here. It should work for all later versions too.

https://github.com/wagtail/wagtail/tree/v2.14/wagtail/core/migrations